src/class/SimpleData.ts

```diff
@@ -50,24 +50,69 @@ export default class SimpleData {
 
     constructor(
         {
-            data = [],
+            url,
+            path,
+            data,
             verbose = false,
             logParameters = false,
-            nbTableItemsToLog = 5
+            nbTableItemsToLog = 5,
+            missingKeyValues = { "null": null, "NaN": NaN, "undefined": undefined },
+            encoding = "utf8"
         }: {
-            data?: SimpleDataItem[]
+            url?: string,
+            path?: string,
+            data?: SimpleDataItem[],
+            encoding?: BufferEncoding,
+            missingKeyValues?: SimpleDataItem,
             verbose?: boolean,
             logParameters?: boolean,
-            nbTableItemsToLog?: number
+            nbTableItemsToLog?: number,
+
         } = {}) {
-        data.length > 0 && checkKeys(data)
 
-        this._data = data
-        this._keys = data[0] && Object.keys(data[0])
+        const allDataArguments = [url, path, data]
+        let nbDataArguments = 0
+        for (const dataArg of allDataArguments) {
+            if (dataArg !== undefined) {
+                nbDataArguments += 1
+            }
+        }
+        if (nbDataArguments === 0) {
+            throw new Error("You must provide either data, url or path.")
+        }
+        if (nbDataArguments > 1) {
+            throw new Error("SimpleData can be created with either data, url or path, but not a combination of them. Provide only one of them.")
+        }
+
+        let incomingData: SimpleDataItem[] = []
+        if (data) {
+            incomingData = data
+        }
+        if (path) {
+            incomingData = loadLocalFile({
+                path: path,
+                verbose: verbose,
+                missingKeyValues: missingKeyValues,
+                encoding: encoding
+            })
+        }
+        if (url) {
+            incomingData = loadUrl()
+        }
+
+        if (incomingData.length > 0) {
+            incomingData.length > 0 && checkKeys(incomingData)
+
+            this._data = incomingData
+            this._keys = incomingData[0] && Object.keys(incomingData[0])
+
+            this.verbose = verbose
+            this.logParameters = logParameters
+            this.nbTableItemsToLog = nbTableItemsToLog
+        } else {
+            throw new Error("Incoming data is empty.")
+        }
```

Comment:
It's best practice to avoid nested code as much as possible. So here, I would do this instead. 
```
if (incomingData.length === 0) {
        throw new Error("Incoming data is empty.")
}
checkKeys(incomingData)

this._data = incomingData
this._keys = Object.keys(incomingData[0])

this.verbose = verbose
this.logParameters = logParameters
this.nbTableItemsToLog = nbTableItemsToLog
```